service/src/test/java/bio/terra/pipelines/service/ImputationServiceMockTest.java

```diff
@@ -53,4 +56,20 @@ void createJobSuccess() {
             TestUtils.TEST_RESULT_URL);
     assertEquals(testUUID, writtenUUID);
   }
+
+  @Test
+  void constructImputationInputsSuccess() {
+    // this inputs definition list includes one service-provided input
+    when(mockPipelinesService.getPipelineInputDefinitions(PipelinesEnum.IMPUTATION_BEAGLE))
+        .thenReturn(TestUtils.TEST_PIPELINE_INPUTS_DEFINITION_LIST);
+
+    // this should add the one service-provided input to the one user-provided input in
+    // testPipelineInputs
+    Map<String, Object> allPipelineInputs =
+        imputationService.constructImputationInputs(
+            PipelinesEnum.IMPUTATION_BEAGLE, testPipelineInputs);
+
+    assertNotNull(allPipelineInputs);
+    assertEquals(2, allPipelineInputs.size());
```

Comment:
a little confused here, doesnt the mock above return `TEST_PIPELINE_INPUTS_DEFINITION_LIST` which has 5 input definitions in there?  where is the 2 coming from?

Reply:
only one of the TEST_PIPELINE_INPUTS_DEFINITION_LIST is service provided. this is an odd combination of these test inputs - when refactoring this method into PipelinesService I hopefully made this test clearer